### PR TITLE
Add databaseless + bug fix

### PR DESCRIPTION
**Pull request by VoidNulll**
Fixed things related to axonOptions, the library selector, added Database-less function and fixed up some more bugs that prevented either the database or the examples from properly functioning.

This fixed bugs related to options, databases, and the library after adding the ability for AxonCore to run without any sort of database.

Database-less is now the default database but JSON is the failover in case someone tries to use a database and it does not work.

### CODE REVIEW

File: src/Database/DBLessService.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * @author VoidNulll
+ *
+ * @class DatabaselessService
+ * @extends DBService
+ */
+class DDBLessService extends DBService {
+    // eslint-disable-next-line no-useless-constructor
+    constructor(axonClient) {
+        super(axonClient);
+        this.databaseless = true;
```
Comment (Khaazz) [inline src/Database/DBLessService.js]: As discussed in DM and commented in the PR

**Comment by Khaazz** [comment]
As discussed in DM.
Here what you have:
- everything stored in memory
- no way to edit GuildConfig (disabled native AxonCore features)
- if LRUCache set to 0: no storage in memory but recreating GuildConfig everytime

Here what I think is better:
- everything stored in mem
- same ability to edit GuildConfig as usual
- if LRUCache set to 0: no storage in memory, but recreating GuildConfig everytime (current behavior)

Basically an in memory solution instead of a DB solution

File: index.d.ts [hunk 2]
```diff
@@ -709,7 +714,7 @@ declare module "axoncore" {
         };
         LIBRARY_TYPES: { ERIS: number, DISCORDJS: 1 };
         LOGGER_TYPES: { DEFAULT: number; CHALK: number; SIGNALE: number; WINSTON: number; };
-        DB_TYPES: { JSON: number; MONGO: number };
+        DB_TYPES: { JSON: number; MONGO: number, DBLESS: number };
```
[the remark this answers is not in the document]
Comment (VoidNulll) [reply index.d.ts]: I guess although it doesn't matter.

**Comment by VoidNulll** [comment]
Two questions
1. I cannot find LRUCache option in the library. Where would it be?
2. IS this what was wanted when you said "in memory" or should the actual memory part be made into the database service itself?

**Comment by Khaazz** [comment]
1 - https://github.com/Khaazz/AxonCore/blob/dev-2.0/src/AxonOptions.js#L30
2 - I like what you did.
You forgot to remove the databaseless option (and any access to the property).
I'd also like a rename of the file to be `InMemoryProvider` (makes more sense than DBLess lol)

**Comment by Khaazz** [comment]
So to sum up about how to correctly handle this, we have 2 possible approaches:
1) The provider / service holds guildsConfigs and axonConfig in a Map (only json data, not the actual objects).
And it is duplicated in Axonclient (current behavior of the framework)

2) The provider / service just get guildConfigs and axonConfig from AxonClient and eventually update it if needed.

You here achieved a correct implementation of solution 2) at first glance.
I am not sure which one of the 2 solutions is the best.
Do we favor semantic or a better memory management.

I don't like duplicating the data but it would be more correct with the rest of the framework though.

**Comment by VoidNulll** [comment]
> Do we favor semantic or a better memory management.

What?

**Comment by Khaazz** [comment]
The best approach would be to have the provider for this case use it's own cache of guildConfig (json format).
That would correctly mimic the system we have with an in memory system.

For now on though, we will stick with your current implementation. Using AxonClient guildConfigs cache directly.
Just waiting for the renaming, and removing entirely the `databaseless` property

File: src/Database/index.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ class DBHandler {
             // No database
             case DB_TYPES.DBLESS:
             default: {
-                const DBLessService = require('./DBLessService').default;
+                const DBLessService = require('./InMemoryProvider').InMemoryProvider;
```
Comment (Khaazz) [inline src/Database/index.js]: Pretty sure this is still `default`.

File: src/Database/InMemoryProvider.js [hunk 4]
```diff
@@ -10,12 +10,7 @@ import GuildConfig from '../Structures/DataStructure/GuildConfig';
  * @class DatabaselessService
```
Comment (Khaazz) [inline src/Database/InMemoryProvider.js]: forgot to rename this one

File: src/Database/InMemoryProvider.js [hunk 5]
```diff
@@ -10,12 +10,7 @@ import GuildConfig from '../Structures/DataStructure/GuildConfig';
...
  * @extends DBService
  */
-class DDBLessService extends DBService {
-    // eslint-disable-next-line no-useless-constructor
-    constructor(axonClient) {
-        super(axonClient);
-        this.databaseless = true;
-    }
+class InMemoryProvider extends DBService {
```
Comment (Khaazz) [inline src/Database/InMemoryProvider.js]: Also rename DBService to DBProvider please (since that's actually the name of the class)

File: index.d.ts [hunk 6]
```diff
@@ -132,12 +132,17 @@ declare module "axoncore" {
     }
 
     // OK
-    export class JsonProvider extends DBProvider {
+    class DBLessService extends DBProvider {
+        public databaseless: true;
```
Comment (Khaazz) [inline index.d.ts]: Typings need to be updated with the latest changes

File: src/Database/index.js (hunk 3 above)
Comment (Khaazz) [reply src/Database/index.js]: Also for the variable name
